facebook-commerce.php

```diff
@@ -3175,6 +3178,26 @@ public function is_product_sync_enabled() {
 		return (bool) apply_filters( 'wc_facebook_is_product_sync_enabled', 'yes' === get_option( self::SETTING_ENABLE_PRODUCT_SYNC, 'yes' ), $this );
 	}
 
+	/**
+ 	 * Return true if (legacy) feed generation is enabled.
+ 	 *
+ 	 * Feed generation for product sync is enabled by default, and generally recommended.
+ 	 * Large stores, or stores running on shared hosting (low resources) may have issues
+ 	 * with feed generation. This option allows those stores to disable generation to
+ 	 * work around the issue.
+ 	 *
+ 	 * Note - this is temporary. In a future release, an improved feed system will be
+ 	 * implemented, which should work well for all stores. This option will not disable
+ 	 * the new improved implementation.
+ 	 *
+ 	 * @since 2.5.0
+ 	 *
+ 	 * @return bool
+ 	 */
+ 	public function is_legacy_feed_file_generation_enabled() {
+ 		return (bool) ( 'yes' === get_option( self::OPTION_LEGACY_FEED_FILE_GENERATION_ENABLED, 'yes' ) );
```

Comment:
Keen for feedback on how this option works. 

- Our immediate, urgent need is to disable the current feed generation (I'm calling this "legacy" in the code/option name).
- We also want to ship an improved feed engine soon - should the option affect that?

To limit the scope of this option, I've decided it should only affect the legacy feed system. When we merge the new improved version, we will need to remove the logic for the option too.

### Alternatives
- Add two options - enable/disable, and "mode". Like a feature flag, so we can ship the new feed engine in parallel and sites can opt-in to the new or old version. This would allow us to ship early and test improvements without them affecting customers. This is most flexible, but might be overkill.
- Add a single "mode" option which includes some combinations of enable/disable and engine mode, e.g. `disable | allow_default | allow_experimental`. This is a feature flag style approach, but with 'stable' (aka default) and 'experimental'; at some point the new engine code would move from `experimental` to `default` and apply to all stores.

Reply:
> We also want to ship an improved feed engine soon - should the option affect that?

No. The control structure for the new feed is in a totally different place.

> Add two options - enable/disable, and "mode". Like a feature flag, so we can ship the new feed engine in parallel and sites can opt-in to the new or old version. This would allow us to ship early and test improvements without them affecting customers. This is most flexible, but might be overkill.

This would confuse people. Besides if the new feed works why have the old one still around?

> Add a single "mode" option which includes some combinations of enable/disable and engine mode, e.g. disable | allow_default | allow_experimental. This is a feature flag style approach, but with 'stable' (aka default) and 'experimental'; at some point the new engine code would move from experimental to default and apply to all stores.

That could be an interesting idea but it should be introduced with an "experimental" build of the plugin and not by a flag.

Reply:
Sounds good - we can look at ways to selectively enable the new engine in future (in line with your and @danielbitzer feedback).